docs/src/api.md

```diff
@@ -0,0 +1,40 @@
+# Function Documentation
+
+```@meta
+CurrentModule = Turing
+```
+
+```@contents
+Pages = ["functions.md"]
+Depth = 5
+```
+
+## Modelling
+
+```@docs
+@model
+@~
```

Comment:
Do we want to include this macro as part of the public facing API @yebai ? i.e. do we want users to see this?

Reply:
My thinking was to include the documentation for anything we `export` -- if someone's using it, I'd assume someone curious will look it up in the documentation. Is there some kind of hard-and-fast guidelines we should follow for what we expose in the public documentation?

Reply:
Ah okay, that seems reasonable. I think a separate discussion is required regarding some of the things that we're exporting.